es6-shim.js

```diff
@@ -1939,9 +1939,7 @@
 
     sign: function sign(value) {
       var number = Number(value);
-      if (number === 0) { return number; }
-      if (Number.isNaN(number)) { return number; }
-      return number < 0 ? -1 : 1;
+      return number < 0 ? -1 : number > 0 ? 1 : number;
```

Comment:
What makes you think this has a meaningful performance impact? The only part that's not trivially optimizable is `Number.isNaN`, and that will be inlined by most every JIT.